idea/src/org/jetbrains/kotlin/idea/intentions/SamConversionToAnonymousObjectIntention.kt

```diff
@@ -39,12 +38,12 @@ class SamConversionToAnonymousObjectIntention : SelfTargetingRangeIntention<KtCa
         val callee = element.calleeExpression ?: return null
         val lambda = getLambdaExpression(element) ?: return null
         val functionLiteral = lambda.functionLiteral
-        val descriptor = (functionLiteral.descriptor as? FunctionDescriptor) ?: return null
         val bindingContext = functionLiteral.analyze()
         val sam = element.getSingleAbstractMethod(bindingContext) ?: return null
 
         val samValueParameters = sam.valueParameters
         val samValueParameterSize = samValueParameters.size
+        val descriptor = bindingContext[BindingContext.DECLARATION_TO_DESCRIPTOR, functionLiteral] as? FunctionDescriptor ?: return null
```

Comment:
You are reusing `bindingContext` here, but in the `applyTo` you are still using `descriptor` extension. What is the reason for that